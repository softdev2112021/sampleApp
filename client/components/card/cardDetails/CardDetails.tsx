export interface CardDetailsProps {
  key: string;
  date: { weekDay: string; date: string };
  content: {
    data: { min: number; max: number; pop: string };
    descr: string;
    icon: string;
  };
}

const CardDetails = (props: CardDetailsProps) => {
  const {
    date: { weekDay, date },
    content: {
      data: { min, max, pop },
      descr,
      icon,
    },
  } = props;
  
  return (
    <div className="col-4 mb-2 d-flex flex-column align-items-center">
      <div className="f-s-12 f-w-800 height-10">{weekDay}/{date}</div>
      <div className="height-50">
        <img src={icon} alt="image" width="64" />
      </div>
      <div className="f-s-18 f-w-800 height-20">{max}<sup>o</sup>C</div>
      <div className="f-s-14 f-w-800 mb-2">{min}</div>
      <i className="fa fa-umbrella">{` ${pop}`}</i>
    </div>
  );
};

export default CardDetails;
