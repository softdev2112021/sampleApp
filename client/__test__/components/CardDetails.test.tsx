import React from 'react';
import renderer from 'react-test-renderer';
import CardDetails from '../../components/card/cardDetails/CardDetails';

it('renders correctly', () => {
  const cardDetailsProps = { 
    date: { weekDay: 'Mon', date: '17/Mar' },
    content: {
      data: { min: 19, max: 20, pop: '0%' },
      descr: 'Cloudly',
      icon: 'icon',
    },
  };

  const tree = renderer
    .create(<CardDetails key={'key'} {...cardDetailsProps}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});