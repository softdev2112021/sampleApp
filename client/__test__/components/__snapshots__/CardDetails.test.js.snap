// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="col-4 mb-2 d-flex flex-column align-items-center"
>
  <div
    className="f-s-12 f-w-800 height-10"
  >
    Mon
    /
    17/Mar
  </div>
  <div
    className="height-50"
  >
    <img
      alt="image"
      src="icon"
      width="64"
    />
  </div>
  <div
    className="f-s-18 f-w-800 height-20"
  >
    20
    <sup>
      o
    </sup>
    C
  </div>
  <div
    className="f-s-14 f-w-800 mb-2"
  >
    19
  </div>
  <i
    className="fa fa-umbrella"
  >
     0%
  </i>
</div>
`;
